proxy/grpcproxy/cluster.go

```diff
@@ -175,3 +181,33 @@ func (cp *clusterProxy) MemberList(ctx context.Context, r *pb.MemberListRequest)
 	resp := (pb.MemberListResponse)(*mresp)
 	return &resp, err
 }
+
+func (ap *authedClusterProxy) MemberAdd(ctx context.Context, r *pb.MemberAddRequest) (*pb.MemberAddResponse, error) {
+	token := getAuthTokenFromClient(ctx)
```

Comment:
these are very invasive; would it be possible to use a unary interceptor instead?